zulipterminal/ui_tools/views.py

```diff
@@ -1169,6 +1169,39 @@ def __init__(self, controller: Any, title: str) -> None:
         super().__init__(controller, widgets, "HELP", popup_width, title)
 
 
+class FileUploadView(PopUpView):
+    def __init__(
+        self,
+        controller: Any,
+        write_box: WriteBox,
+        title: str,
+    ) -> None:
+        self.controller = controller
+        self.model = controller.model
+        self.write_box = write_box
+        max_cols, max_rows = controller.maximum_popup_dimensions()
+        self.predefined_text = urwid.Text("Location : ")
+        self.file_location_edit = urwid.Edit()
+        columns = [self.predefined_text, self.file_location_edit]
+        super().__init__(
+            controller,
+            columns,
+            "FILE_UPLOAD",
+            max_cols,
+            title,
+        )
+
+    def _handle_file_upload(self, file_location: str) -> None:
+        self.uri = self.model.get_file_upld_uri(file_location)
+        file_name = file_location.split("/")[-1]
+        self.write_box.insert_uri(file_name, self.uri)
+
+    def keypress(self, size: urwid_Size, key: str) -> str:
+        if is_command_key("FILE_UPLOAD", key):
```

Comment:
You're overriding the behavior of the keypress here, so you should be able to change the key we use here quite easily?